### ISSUE TITLE
Uosc/Playlist to Show Filename Instead of Title Name

### ISSUE THREAD

**Issue by Cananbolt** [tomasklaen/uosc#888]
As the title states, currently, uosc/playlist menu shows the title name, I was wondering if it was possible for the menu to show the filename instead. This problem arises when you are on the opened file.

### For example, I have a playlist of 3

Filename would be something like...

[Group] Yoyo - 01
[Group] Yoyo - 02
[Group] Yoyo - 03

### Now if I am currently watching...

[Group] Yoyo - 01

...and if it has a title the uosc/playlist menu would show...

Yoyo

...now when I move on to another video in the playlist it will keep showing the title, instead of the filename.

**Comment by Cananbolt** [comment]
Thank you for breaking it down. If I may take this further, would it be possible for network files to have title name (uosc default) and then for local files it would be the filename. **Only for the uosc/playlist menu**.

I tried what you said about the youtube videos and yes, that is absolutely correct, you definitely don't want to see url's instead of the actual title.

However, when it comes to local files, in those cases, normally, the filename is preferred.

Edit: I actually found where to change between filename and title name

```lua
bind_command('playlist', create_self_updating_menu_opener({
	title = 'Playlist',
	type = 'playlist',
	list_prop = 'playlist',
	serializer = function(playlist)
		local items = {}
		for index, item in ipairs(playlist) do
			local is_url = item.filename:find('://')
			local item_title = type(item.title) == 'string' and #item.title > 0 and item.title or false
			items[index] = {
				title = item_title or (is_url and item.filename or serialize_path(item.filename).basename),
				hint = tostring(index),
				active = item.current,
				value = index,
			}
		end
		return items
``` 
Where you can change between filename and title is
```lua
				title = item_title or (is_url and item.filename or serialize_path(item.filename).basename),

``` 
Now we can change the `item_title` to `filename` if we desire.

However, as you pointed out this results in URLs in the uosc/playlist and I replicated this myself when I changed item_title to filename in the code above.

This brings up my initial point, is it possible to differentiate between local and network files. Maybe if the filename has https:// it would keep everything above the same but for filenames that does not have https:// it will use the filename instead.

### Tldr: Can we make it so that network files use title name (uosc default) whearas local files will use filenames.

**Comment by Cananbolt** [comment]
Wow, ChatGPT is incredible!

```lua
bind_command('playlist', create_self_updating_menu_opener({
    title = 'Playlist',
    type = 'playlist',
    list_prop = 'playlist',
    serializer = function(playlist)
        local items = {}
        for index, item in ipairs(playlist) do
            local is_url = item.filename:find('://')
            local title
            
            if is_url then
                if type(item.title) == 'string' and #item.title > 0 then
                    title = item.title
                else
                    title = item.filename
                end
            else
                title = serialize_path(item.filename).basename
            end
            
            items[index] = {
                title = title,
                hint = tostring(index),
                active = item.current,
                value = index,
            }
        end
        return items
    end,
    on_select = function(index)
        mp.commandv('set', 'playlist-pos-1', tostring(index))
    end,
    on_move_item = function(from, to)
        mp.commandv('playlist-move', tostring(math.max(from, to) - 1), tostring(math.min(from, to) - 1))
    end,
    on_delete_item = function(index)
        mp.commandv('playlist-remove', tostring(index - 1))
    end,
```
I have not done any thorough testing, only tested it once. But this does exactly what I want it to. For filenames that **does not have** '://' it will use filenames and for filenames that **does have** '://' it will use filenames. This is done through ChatGPT so I am unsure of its consistency, however in initial testing it is working for me.

**Comment by Cananbolt** [comment]
@christoph-heinrich

Thank you for updating the code, it is so much cleaner!

I have tried it as well, and as intended local files show the filename whereas protocols show the title name.

@tomasklaen 

Thank you both for fulfilling my request!

**Comment by christoph-heinrich** [comment]
@tomasklaen I'm not sure.
There is no way to differentiate between titles set by the playlist and titles acquired by opening the file, and in cases where the playlist provides a title it's probably desirable to show that title. Otherwise it seems like a good idea.

**Comment by tomasklaen** [comment]
Yeah you're right, ignoring titles would be a bad idea.
